charts/yourls/templates/NOTES.txt

```diff
@@ -89,19 +89,15 @@ You can access Apache Prometheus metrics following the steps below:
 {{- include "common.warnings.rollingTag" .Values.image }}
 {{- include "common.warnings.rollingTag" .Values.metrics.image }}
 {{- include "common.warnings.rollingTag" .Values.volumePermissions.image }}
-{{- $passwordValidationErrors := list -}}
 {{- if not .Values.yourls.existingSecret -}}
-    {{- $secretName := include "yourls.secretName" . -}}
-    {{- $requiredYOURLSPassword := dict "valueKey" "yourls.password" "secret" $secretName "field" "password" "context" $ -}}
-    {{- $requiredYOURLSPasswordError := include "common.validations.values.single.empty" $requiredYOURLSPassword -}}
-    {{- $passwordValidationErrors = append $passwordValidationErrors $requiredYOURLSPasswordError -}}
-{{- end }}
-{{- if .Values.mysql.enabled }}
-    {{- $mysqlSecretName := include "yourls.databaseSecretName" . -}}
-    {{- $mysqlPasswordValidationErrors := "OK" -}}
-{{/* TODO: Validate MySQL
-    {{- $mysqlPasswordValidationErrors := include "common.validations.values.mariadb.passwords" (dict "secret" $mysqlSecretName "subchart" true "context" $) -}}
-*/}}
-    {{- $passwordValidationErrors = append $passwordValidationErrors $mysqlPasswordValidationErrors -}}
+  {{- $secretName := include "yourls.secretName" . -}}
+  {{- $requiredPasswords := list -}}
+
+  {{- if not (empty .Values.yourls.username) -}}
+    {{- $requiredPassword := dict "valueKey" "yourls.password" "secret" $secretName "field" "password" -}}
+    {{- $requiredPasswords = append $requiredPasswords $requiredPassword -}}
+  {{- end -}}
+
+  {{- $yourlsPasswordValidationErrors := include "common.validations.values.multiple.empty" (dict "required" $requiredPasswords "context" $) -}}
+  {{- include "common.errors.upgrade.passwords.empty" (dict "validationErrors" $yourlsPasswordValidationErrors "context" $) -}}
 {{- end }}
-{{- include "common.errors.upgrade.passwords.empty" (dict "validationErrors" $passwordValidationErrors "context" $) -}}
```

Comment:
The password validation function should stay outside of the actual validation for maintainability.